### ISSUE TITLE
Get rid of `${maven.multiModuleProjectDirectory}`

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#42430]
This was apparently an internal property and it got removed from mvnd.

We put a hack in place here: https://github.com/quarkusio/quarkus/pull/41648 but we should get rid of it and use `${session.rootDirectory}` instead.

This property was introduced in https://issues.apache.org/jira/browse/MNG-7038 and has been backported to 3.9.2: https://maven.apache.org/docs/3.9.2/release-notes.html .

**Comment by maxandersen** [comment]
this just affects quarkus own build, right ? as this would mean mvn as found on rhel/ubi 8 wouldn't be able to build I reckon.

**Comment by gsmet** [comment]
Yes.

**Comment by gsmet** [comment]
Actually, the situation is a bit more complex than that, I should have read the fine prints of the 3.9.2 release notes.

While these properties are available across the board in Maven 4, in Maven 3.9, they are restricted to the Maven config, they are not usable in a POM file:

> It is important to mention, that those two new properties, are only working within the maven.config file and via the command line. They are NOT usable as configuration properties within your plugin configuration in your pom.xml file.

We will have to wait for Maven 4 before cleaning this up.

**Comment by manofthepeace** [comment]
Yes, this is why we went with the hack in the current solution, and not simply doing the "replace all"
